core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -312,7 +312,15 @@ class LogManager(logDirs: Seq[File],
       val logDirAbsolutePath = dir.getAbsolutePath
       var hadCleanShutdown: Boolean = false
       try {
-        val pool = Executors.newFixedThreadPool(numRecoveryThreadsPerDataDir)
+        val pool = Executors.newFixedThreadPool(numRecoveryThreadsPerDataDir, new ThreadFactory {
```

Comment:
line#495 (`shutdown`) creates thread poll also. Does it need a better naming?

Reply:
Yes, thanks! I've now factored out a common method.